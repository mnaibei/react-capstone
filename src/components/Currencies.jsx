import { useState, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { getCurrencies } from "../redux/currencies/currencySlice";
import "../styles/currency.css";
import { useNavigate } from "react-router-dom";

const Currencies = () => {
  const { currencies } = useSelector((store) => store.currencies);
  const dispatch = useDispatch();
  const navigate = useNavigate();

  // Add state for search query
  const [searchQuery, setSearchQuery] = useState('');

  useEffect(() => {
    dispatch(getCurrencies());
  }, [dispatch]);

  const handleDetails = (currency) => {
    navigate(`/details/${currency.name.toLowerCase()}`, { state: { currency } });
  };

  // Filter currencies based on search query
  const filteredCurrencies = currencies.filter((currency) =>
    currency.name.toLowerCase().includes(searchQuery.toLowerCase())
  );

  return (
    <>
      {/* Add search input field */}
      <input
        type="text"
        placeholder="Search coin..."
        value={searchQuery}
        onChange={(e) => setSearchQuery(e.target.value)}
      />

      {filteredCurrencies.map((currency) => (
        <div
          className="currency-container"
          key={currency.id}
          onClick={() => {
            handleDetails(currency);
          }}
        >
          <h2 className="symbol">{currency.symbol}</h2>
        </div>
      ))}
    </>
  );
};

export default Currencies;
