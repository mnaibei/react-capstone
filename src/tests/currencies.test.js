import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import Currencies from '../components/Currencies';
import Store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Currencies', () => {
  test('Render Currencies Component', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Provider store={Store}>
          <Currencies />
        </Provider>,
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
