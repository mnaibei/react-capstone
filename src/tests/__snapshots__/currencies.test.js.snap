// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currencies Render Currencies Component 1`] = `
Array [
  <h1
    className="title"
  >
    CryptoCurrency Tracker
  </h1>,
  <div
    className="search"
  >
    <input
      className="search-input"
      onChange={[Function]}
      placeholder="Search coin..."
      type="text"
      value=""
    />
  </div>,
  <div
    className="currency-container"
  />,
  ",",
]
`;
